packages/block-library/src/heading/heading-checker.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * WordPress dependencies
+ */
+import { speak } from '@wordpress/a11y';
+import { __ } from '@wordpress/i18n';
+import { Notice } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { computeOutlineHeadings } from '../../../editor/src/components/document-outline/index.js';
+
+export const HeadingChecker = ( { blocks = [], selectedLevel, selectedHeadingId } ) => {
```

Comment:
How do you feel about renaming it to `HeadingLevelChecker` to better reflects what it validates?

The name of the file could be `heading/level-checker.js` in that case.